wasmer2/wasmer2Executor.go

```diff
@@ -112,8 +106,29 @@ func (wasmerExecutor *Wasmer2Executor) NewInstanceFromCompiledCodeWithOptions(
 	compiledCode []byte,
 	options executor.CompilationOptions,
 ) (executor.Instance, error) {
-	// return NewInstanceFromCompiledCodeWithOptions(compiledCode, options)
-	return nil, errors.New("NewInstanceFromCompiledCodeWithOptions not implemented")
+	var c_instance *cWasmerInstanceT
+
+	if len(compiledCode) == 0 {
+		var emptyInstance = &Wasmer2Instance{cgoInstance: nil}
+		return emptyInstance, newWrappedError(ErrInvalidBytecode)
+	}
+
+	cOptions := unsafe.Pointer(&options)
+	var compileResult = cWasmerInstanceFromCache(
+		wasmerExecutor.cgoExecutor,
+		&c_instance,
+		(*cUchar)(unsafe.Pointer(&compiledCode[0])),
+		cUint32T(len(compiledCode)),
+		(*cWasmerCompilationOptions)(cOptions),
+	)
+
+	if compileResult != cWasmerOk {
+		var emptyInstance = &Wasmer2Instance{cgoInstance: nil}
+		return emptyInstance, newWrappedError(ErrFailedInstantiation)
```

Comment:
Same as above, `nil` instead of `emptyInstance`.

Reply:
done